frontend/src/App.tsx

```diff
@@ -1,31 +1,43 @@
 import React from "react";
+// firebase imports
 import {
   createBrowserRouter,
   createRoutesFromElements,
   Route,
   RouterProvider,
-  Navigate,
+  Navigate
 } from "react-router-dom";
+// pages
 import Signup from "./pages/signup/Signup";
 import Login from "./pages/login/Login";
 import Password from "./pages/password/Password";
 import Dashboard from "./pages/dashboard/Dashboard";
+// hooks
+import { useAuthContext } from "./hooks/useAuthContext";
+// styles
 import "./App.css";
 
+function App(): JSX.Element {
+  const { user } = useAuthContext();
 
-const router = createBrowserRouter(
-  createRoutesFromElements(
-    <React.Fragment>
-      <Route path="/signup" element={<Signup />} />
-      <Route path="/login" element={<Login />} />
-      <Route path="/password" element={<Password />} />
-      <Route path="/dashboard" element={<Dashboard />} />
-      <Route path="/" element={<Navigate to="/login" replace />} />
-    </React.Fragment>
-  )
-);
+  const router = createBrowserRouter(
+    createRoutesFromElements(
+      <React.Fragment>
+        <Route
+          path="/"
+          element={user ? <Navigate to="/dashboard" replace /> : <Navigate to="/login" replace />}
+        />
+        <Route path="/signup" element={user ? <Navigate to="/dashboard" replace /> : <Signup />} />
+        <Route path="/login" element={user ? <Navigate to="/dashboard" replace /> : <Login />} />
+        <Route
+          path="/password"
+          element={user ? <Navigate to="/dashboard" replace /> : <Password />}
+        />
+        <Route path="/dashboard" element={user ? <Dashboard /> : <Navigate to="/login" replace />} />
+      </React.Fragment>
+    )
+  );
```

Comment:
You are repeating a lot of code and with every added route you would need to do these repetitions again and again, please find another method (You can refer to the documentation https://reactrouter.com/en/main/start/tutorial#nested-routes)